plugin.go

```diff
@@ -0,0 +1,391 @@
+// Copyright 2013 go-dockerclient authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package docker
+
+import (
+	"encoding/json"
+	ioutil "io/ioutil"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/net/context"
+)
+
+// PluginPrivilege represents a privilege for a plugin.
+type PluginPrivilege struct {
+	Name        string   `json:"Name,omitempty" yaml:"Name,omitempty" toml:"Name,omitempty"`
+	Description string   `json:"Description,omitempty" yaml:"Description,omitempty" toml:"Description,omitempty"`
+	Value       []string `json:"Value,omitempty" yaml:"Value,omitempty" toml:"Value,omitempty"`
+}
+
+// InstallPluginOptions .This is a TBD Comments.
+//
+// See https://goo.gl/kaOHGw for more details.
+type InstallPluginOptions struct {
+	Remote  string
+	Name    string
+	Plugins []PluginPrivilege
+	Context context.Context
+}
+
+// InstallPlugins returns a slice of containers matching the given criteria.
+//
+// See https://goo.gl/kaOHGw for more details.
+func (c *Client) InstallPlugins(opts InstallPluginOptions, auth AuthConfiguration) error {
+	params := make(url.Values)
+	params.Set("remote", opts.Remote)
+	if opts.Name != "" {
+		params.Set("name", opts.Name)
+	}
+	path := "/plugins/pull?" + queryString(params)
```

Comment:
Since you're using ``queryString()``, you can add struct tags to fields on ``InstallPluginOptions`` and just do:

```go
path := "/plugins/pull?" + queryString(opts)
```

``InstallPluginOptions`` would look something like:

```go
type InstallPluginOptions struct {
	Remote  string
	Name    string
	Plugins []PluginPrivilege `qs:"-"`
	Context context.Context  
}
```